docs/adrs/0002-dedicated-timdex-record-field-methods.md

```diff
@@ -0,0 +1,101 @@
+
+Moving to a method-per-field approach will also allow for more granular logging.
+
+## Decision
+
+### 1- Refactor the base `Transformer` class to have all `TimdexRecord` fields as dedicated methods
+
+At the time of writing, the `TimdexRecord` has 30 fields:
+
+  * REQUIRED
+    * `source`
+    * `source_link`
+    * `timdex_record_id`
+    * `title`
+
+  * OPTIONAL
+    * `alternate_titles`
+    * `call_numbers`
+    * `citation`
+    * `content_type`
+    * `contents`
+    * `contributors`
+    * `dates`
+    * `edition`
+    * `file_formats`
+    * `format`
+    * `funding_information`
+    * `holdings`
+    * `identifiers`
+    * `languages`
+    * `links`
+    * `literary_form`
+    * `locations`
+    * `notes`
+    * `numbering`
+    * `physical_description`
+    * `publication_frequency`
+    * `publication_information`
+    * `related_items`
+    * `rights`
+    * `subjects`
+    * `summary`
+
+The base `Transformer` class will have a dedicated method for each of these 30 `TimdexRecord` fields, providing a clear and modular approach to defining logic for setting those fields. 
+
+The required fields will be an `@abstractmethod`, meaning any extending classes will be required to define them.
+
+The rest of the optional fields will receive an `@optional_field("<FIELD_NAME>")` decorator that will identify them as a) methods meant to parsing data for a given field, and b) the name of that field.  Using a decorator provides a few of benefits:
+
+1. It can be programmatically determined which transformations define logic, for which fields, even into inheritance
+2. Hard-coding method-to-field invocations, or using method naming conventions, are not needed
+3. Decorators could eventually pass more information to methods if needed (e.g. state)
+4. Self-documenting that methods have an explicit purpose
+
+From there, transformations will extend `Transformer` and override methods for specific fields, with logic specific to that transformation.  Multiple inheritance is also simplified in this approach, where extending classes continue to refine the behavior for a specific target field.
+
+### 2- Refactor pre-existing transformations to use this new approach
+
+With the new structure in place, refactor pre-existing source transformations to use it.  This may include revisiting some of the transformation inheritance if there is an opportunity for intermediate transformation classes now that an intermediate class could theoretically handle only one, two, or a handful of fields. 
```

Comment:
I'm not totally clear on what is meant by `intermediate transformation class`, an example could be helpful

Reply:
I am curious about this as well!

Reply:
Added a bit about transformation inheritance in this commit: https://github.com/MITLibraries/transmogrifier/pull/105/commits/d93fa26cbaccdadcb31fa534a3338214177440c0

And, removed some of this "intermediate" language which is probably pre-mature optimization / considerations at this point.  What I do think stands, is how this would make pretty straight-forward inheritance much easier to reason about and implement.

Reply:
Realizing I did _not_ remove the "intermediate" language, but hoping maybe it moves the needle a bit in what they could be.  In short: you could _theoretically_ have transformation classes that pinpoint just a couple of fields, and then "mix" those in when defining a new transformation, being confident you'd only pickup their methods for a couple of fields.